packages/ui/Scrollable/Scrollable.js

```diff
@@ -693,79 +699,74 @@ class ScrollableBase extends Component {
-					if (position === 'before' && curPos <= 0) { // Already on the beginning egde
-						applyOverscrollEffect(orientation, 'before', type, ratio);
-						newType = overscrollTypes.none;
-						newRatio = 0;
-					} else if (position >= 'after' && curPos >= maxPos) { // Already on the ending egde
-						applyOverscrollEffect(orientation, 'after', type, ratio);
-						newType = overscrollTypes.none;
-						newRatio = 0;
-					}
-				} else if (type === overscrollTypes.sticking) {
-					if (position === 'before' && curPos <= 0) { // Already on the beginning egde
-						applyOverscrollEffect(orientation, 'before', type, ratio);
-					} else if (position === 'after' && curPos >= maxPos) { // Already on the ending egde
-						applyOverscrollEffect(orientation, 'after', type, ratio);
-					}
+				maxPos = this.getScrollBounds()[isVertical ? 'maxTop' : 'maxLeft'],
+				{none, scrolling} = overscrollTypes;
+
+			if (ratio > 0 && type === scrolling) {
+				if (position === 'before' && curPos <= 0) { // Already on the beginning egde
+					applyOverscrollEffect(orientation, 'before', type, ratio);
+					this.setOverscrollStatus(orientation, position, none, 0);
+				} else if (position === 'after' && curPos >= maxPos) { // Already on the ending egde
+					applyOverscrollEffect(orientation, 'after', type, ratio);
+					this.setOverscrollStatus(orientation, position, none, 0);
+				} else {
+					this.setOverscrollStatus(orientation, position, type, ratio);
 				}
-
-				this.setOverscrollStatus(orientation, newType, newRatio);
 			}
 		}
 	}
 
-	updateOverscrollEffect = (orientation, position, type, ratio = null) => {
+	// This should be called by user interaction
+	updateOverscrollEffect = (orientation, targetPos, type, ratio = null) => {
 		const applyOverscrollEffect = this.props.applyOverscrollEffect;
 
 		if (applyOverscrollEffect && this.getScrollabilities(orientation)) {
 			const
 				isVertical = (orientation === 'vertical'),
 				curPos = isVertical ? this.scrollTop : this.scrollLeft,
-				maxPos = this.getScrollBounds()[isVertical ? 'maxTop' : 'maxLeft'];
+				maxPos = this.getScrollBounds()[isVertical ? 'maxTop' : 'maxLeft'],
+				{none, scrolling} = overscrollTypes;
 			let
-				newType = type,
-				newRatio = (ratio === null) ? this.getOverscrollRatio(orientation, position) : ratio;
-
-			if (newRatio > 0) {
-				if (type === overscrollTypes.scrolling) {
-					if (position <= 0 && curPos <= 0) { // Already on the beginning egde
-						applyOverscrollEffect(orientation, 'before', type, newRatio);
-						newType = overscrollTypes.none;
-						newRatio = 0;
-					} else if (position >= maxPos && curPos >= maxPos) { // Already on the ending egde
-						applyOverscrollEffect(orientation, 'after', type, newRatio);
-						newType = overscrollTypes.none;
-						newRatio = 0;
+				showEffectBefore = false,
+				showEffectAfter = false;
+
+			ratio = (ratio === null) ? this.getOverscrollRatio(orientation, targetPos) : ratio;
```

Comment:
Please do not reassign the value to a parameter of a function.
Reference:
https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#31-reassigning-a-defined-parameter-while-also-mentioning-arguments-in-the-body-in-sloppy-mode-only-typical-example